addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -0,0 +1,498 @@
+                                    contentController.renderListing( standardAccept, sk, path + "/", baseUri,
+                                                                     uriFormatter, eventMetadata );
+
+                            response =
+                                    formatOkResponseWithEntity( content, acceptInfo.getRawAccept(), builderModifier );
+                        }
+                        catch ( final IndyWorkflowException e )
+                        {
+                            logger.error(
+                                    String.format( "Failed to render content listing: %s from: %s. Reason: %s", path,
+                                                   name, e.getMessage() ), e );
+                            response = formatResponse( e, builderModifier );
+                        }
+                    }
+                    else
+                    {
+                        logger.info( "RETURNING: retrieval of content: {}:{}", sk, path );
+                        // open the stream here to prevent deletion while waiting for the transfer back to the user to start...
+                        InputStream in = item.openInputStream( true, eventMetadata );
+                        final Response.ResponseBuilder builder = Response.ok( new TransferStreamingOutput( in ) );
+                        setInfoHeaders( builder, item, sk, path, true, contentController.getContentType( path ),
+                                        contentController.getHttpMetadata( item ) );
+                        if ( builderModifier != null )
+                        {
+                            builderModifier.accept( builder );
+                        }
+                        response = builder.build();
+                        // generating .http-metadata.json for npm group retrieve to resolve header requirements
+                        if ( eventMetadata.get( STORAGE_PATH ) != null && StoreType.group == st )
+                        {
+                            generateHttpMetadataHeaders( item, request, response );
+                        }
+                    }
+                }
+                finally
+                {
+                    if ( handleLocking )
+                    {
+                        item.unlock();
+                    }
+                }
+            }
+            catch ( final IOException | IndyWorkflowException e )
+            {
+                logger.error( String.format( "Failed to download artifact: %s from: %s. Reason: %s", path, name,
+                                             e.getMessage() ), e );
+                response = formatResponse( e, builderModifier );
+            }
+        }
+
+        logger.info( "RETURNING RESULT: {}:{}", sk, path );
+        return response;
+    }
+
+    private void generateNPMContentsFromTransfer( final Transfer transfer, final EventMetadata eventMetadata,
+                                                  Response response,
```

Comment:
Seems this "response" param is not used in this method, but just be set with some value.

Reply:
@gli, I’ve made the method to throw the IndyWorkflowException and let the doCreate method to catch that and do a response with these kinds of error, yep, this should be the right way to resolve, thanks.